### PR TITLE
Feature: (ISA-150) Focus catalogue layer click

### PR DESCRIPTION
**Pull request by JoshGx2000**
Solves https://jira.its.utas.edu.au/browse/ISA-150

Some new events have been added, with code to open a layer in the catalogue and focus that layer on the map when it is selected in the layers search omnibar.

Screen Shot 2022-06-06 at 10 24 46 am
Screen Shot 2022-06-06 at 10 25 00 am
Screen Shot 2022-06-06 at 10 25 23 am

### CODE REVIEW

File: frontend/src/cljs/imas_seamap/events.cljs [hunk 1]
```diff
@@ -511,11 +511,33 @@
...
+    {:db       db
+     :put-hash (encode-state db)}))
+
+(defn catalogue-add-nodes-to-layer
+  "Opens nodes in catalogue along path to specified layer"
+  [{:keys [db]} [_ group layer tab categories]]
+  (let [sorting-info (:sorting db)
+        node-ids   (reduce
+                    (fn [node-ids category]
+                      (let [sorting-id (or (get-in sorting-info [category (category layer) 1]) "nil")
```
Comment (markhepburn) [inline frontend/src/cljs/imas_seamap/events.cljs:532]: get-in can return a default (not-found) value: https://clojuredocs.org/clojure.core/get-in

File: frontend/src/cljs/imas_seamap/map/utils.cljs [hunk 2]
```diff
@@ -279,3 +279,56 @@
         basemap-groups (concat basemap-groups ungrouped-layers-groups)
         basemap-groups (sort-by (juxt #(or (:sort_key %) "zzzzzzzzzz") :id) basemap-groups)]
     basemap-groups))
+
+;; We need a better place for these - our goal is to genericize the catalogues
```
Comment (markhepburn) [inline frontend/src/cljs/imas_seamap/map/utils.cljs:283]: Indeed -- I might write a ticket for that in fact, if you're idle!

**Review by markhepburn — CHANGES_REQUESTED** [review]
Couple of things that can be tidied up -- I'll write a ticket (if there isn't one) about handling arbitrary categories.  Otherwise looks good thanks.

File: frontend/src/cljs/imas_seamap/events.cljs [hunk 3]
```diff
@@ -511,11 +511,33 @@
...
-        updated-nodes (if (contains? nodes nodeid) (disj nodes nodeid) (conj nodes nodeid))
+        updated-nodes (if (nodes nodeid) (disj nodes nodeid) (conj nodes nodeid))
         db            (assoc-in db [:display :catalogue group :expanded] updated-nodes)]
     {:db       db
      :put-hash (encode-state db)}))
 
+(defn catalogue-add-node [{:keys [db]} [_ group nodeid]]
+  (let [nodes         (get-in db [:display :catalogue group :expanded])
+        updated-nodes (conj nodes nodeid)
+        db            (assoc-in db [:display :catalogue group :expanded] updated-nodes)]
```
[the remark this answers is not in the document]
Comment (JoshGx2000) [reply frontend/src/cljs/imas_seamap/events.cljs:522]: Right you are. Honestly slipped my mind.

File: frontend/src/cljs/imas_seamap/events.cljs (hunk 1 above)
Comment (JoshGx2000) [reply frontend/src/cljs/imas_seamap/events.cljs:532]: Can't believe this slipped my mind too.

File: frontend/src/cljs/imas_seamap/map/utils.cljs (hunk 2 above)
Comment (JoshGx2000) [reply frontend/src/cljs/imas_seamap/map/utils.cljs:283]: Good idea – it's not like I have anything better to do 😉

**Pull request merged**